corehq/apps/sms/forms.py

```diff
@@ -616,6 +634,17 @@ def __init__(self, data=None, domain=None, is_previewer=False, *args, **kwargs):
             "add_window_method": "addSMSConversationTime",
         }
 
+    def set_admin_email_choices(self, data, kwargs):
+        email_choices = []
+        if 'initial' in kwargs and 'sms_worker_registration_alert_emails' in kwargs['initial']:
+            email_choices = kwargs['initial']['sms_worker_registration_alert_emails']
+        elif data:
+            email_choices = data.getlist('sms_worker_registration_alert_emails')
```

Comment:
is this if/else for GET and POST?

Reply:
Yes. Is there a better way of doing this in a form?

Reply:
Was just trying to understand why there is a branching here.

Checking that if there are initial values or not, does not reveal the actual need for the check for request.
I was thinking of
```
if request.method == 'GET':
    email_choices = kwargs['initial']['sms_worker_registration_alert_emails']
elif request.method == 'POST' and data:
    email_choices = data.getlist('sms_worker_registration_alert_emails')
```
unless that breaks the forms.
Though, I am fine with the current implementation as well.

Reply:
@proteusvacuum I must agree with @mkangia ; it's not clear why the branching exists. I'd even be satisfied with a comment if the `request.method == 'GET'` is too hard to get in.

Reply:
It appears from [Django docs](https://docs.djangoproject.com/en/2.2/ref/forms/api/#django.forms.Form.has_changed) that a form can be instantiated with both `data` and `initial`. Maybe that `elif data:` should be an `if data:`?